lib/Basics/SourceLocation.h

```diff
@@ -23,34 +23,84 @@
 
 #pragma once
 
+#include <cstdint>
 #include <iosfwd>
+#include <source_location>
+#include <string_view>
 
 namespace arangodb::basics {
+consteval auto stripPrefix(const char* pathStr) -> const char*;
 
-// Poor-man's replacement for std::source_location, until we get C++20.
+// This is mostly a custom type around std::source_location with the same
+// interface; However, it changes the filename's path, so it's relative to the
+// git root directory.
 struct SourceLocation {
  private:
-  char const* _fileName;
-  int _line;
+  char const* _fileName{};
+  char const* _functionName{};
+  std::uint_least32_t _line{};
+  std::uint_least32_t _column{};
 
  public:
+  consteval static auto current(
+      std::source_location loc = std::source_location::current())
+      -> SourceLocation {
+    return SourceLocation{loc};
+  }
+
+  // We can allow this if we need to, but this way it prevents accidental usage
+  // of uninitialized SourceLocations.
   SourceLocation() = delete;
-  // cppcheck-suppress unknownMacro
-  constexpr SourceLocation(decltype(_fileName) file,
-                           decltype(_line) line) noexcept
-      : _fileName(file), _line(line) {}
+
+  consteval explicit SourceLocation(
+      std::source_location const location) noexcept
```

Comment:
nit: as we are passing by value here, I don't think that the `const` actually matters.
```suggestion
      std::source_location location) noexcept
```